### PR TITLE
Operator Deployment Example with KUDO/NGINX

### PR DESCRIPTION
**Pull request by austinchambers**
# https://kudo.dev/docs/developing-operators/getting-started.html#package-structure

# Prerequisites

The KUDO kubectl plugin. 
```s
brew tap kudobuilder/tap
brew install kudo-cli
```
or 
```s
kubectl krew install kudo
```

# Setup
1. Create a VPC in GCE that is airgapped (or close enough, only allowing 22 and 8800 ingress)
2. Create an instance in GCE that uses the VPC. 
3. Install to the airgapped VM
```s
curl -sSL https://k8s.kurl.sh/kudo-nginx-operator-austin | sudo bash
```
4. Verify you have kudo instances generated: 
```s
kubectl kudo get instances 
```
5. Verfy you have two kudo pods: 
```s
kubectl get pods 
```

# How I generated the Operator YAML using KUDO
1. Initialize KUDO and create Kudo manifest (including CRDs). Or use what is in repo 
```s
kubectl kudo init --dry-run --output yaml > ./manifests/kudo.yaml
```
2. Extract the YAML from those manifests (for some reason, kudo doesn't have dry-run install). Replace first three vars with appropriate values
```s
OPERATOR_FILENAME=kudo-operator.yaml
NAMESPACE=kudo
OPERATOR_NAME=first-operator

cd manifests
kubectl get operators.kudo.dev $OPERATOR_NAME -n $NAMESPACE --export -n $NAMESPACE -o yaml > $OPERATOR_FILENAME && echo --- >> $OPERATOR_FILENAME
TMP_OPVERNAME=$(kubectl get operatorversions.kudo.dev -n kudo -o jsonpath="{.items[?(@.spec.operator.name==\"${OPERATOR_NAME}\")].metadata.name}")
kubectl get operatorversions.kudo.dev ${TMP_OPVERNAME} --export -n $NAMESPACE -o yaml >> $OPERATOR_FILENAME && echo --- >> $OPERATOR_FILENAME
kubectl get instances.kudo.dev ${OPERATOR_NAME}-instance --export -n $NAMESPACE -o yaml >> $OPERATOR_FILENAME && echo --- >> $OPERATOR_FILENAME
cd ..
```
3. Remove kudo and all yaml from the cluster
```s
kubectl kudo init --dry-run --output yaml | kubectl delete -f -
```

### CODE REVIEW

**Review by aj-jester — APPROVED** [review]
I need to read up on kudo some more but this looks great.

File: kudo-nginx-operator/manifests/support-bundle.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+apiVersion: troubleshoot.replicated.com/v1beta1
```
Comment (dexhorthy) [inline kudo-nginx-operator/manifests/support-bundle.yaml:1]: Not a blocker, but I'd encourage to make these more minimal, at least removing things like these defaults, if not making them actually pull the right logs

```
    - secret:
        name: illmannered-cricket-mysql
        namespace: default
        key: mysql-password
    - logs:
        selector:
          - name=nginx-ingress-microk8s
        namespace: default
        limits:
          maxAge: 30d
          maxLines: 10000
```

File: kudo-nginx-operator/manifests/preflight.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+apiVersion: troubleshoot.replicated.com/v1beta1
```
Comment (dexhorthy) [inline kudo-nginx-operator/manifests/preflight.yaml:1]: same for preflight, we should consider removing the defaults that never pass and are more for demonstration that anything else

File: kudo-nginx-operator/manifests/support-bundle.yaml (hunk 1 above)
Comment (austinchambers) [reply kudo-nginx-operator/manifests/support-bundle.yaml:1]: Yeah, it's a good point. My process is to pull from the default yaml and work from that. I think we should maybe make that simpler.

**Pull request merged**